packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -1514,6 +1514,24 @@ export default class RestApi {
         res.sendStatus(200)
       })
     )
+
+    api.delete(
+      '/:collection(groups)/:id',
+      wrap(async (req, res) => {
+        const { id } = req.params
+        try {
+          await app.deleteGroup(id)
+          res.sendStatus(204)
+        } catch (error) {
+          json({ error: error.message })
+          if (noSuchObject.is(error)) {
+            res.status(404)
+          } else {
+            res.status(500)
+          }
+        }
```

Comment:
In case of error, you do not send a response to the client. This means that the client will "load" indefinitely